commit 912b5ba94e3a9236e8f4998e6f1b4168ee57e2b0 (refs/changes/69/2969/3)
Author: Rob Johnston <rob.johnston@joyent.com>
Date:   2017-11-28T19:22:00+00:00 (1 year, 10 months ago)
    
    OS-6448 fmdump(1m) should be more resilient in the face of missing message content

diff --git a/usr/src/cmd/fm/fmdump/common/fault.c b/usr/src/cmd/fm/fmdump/common/fault.c
index a8fa104a20..f4ccfd2ef7 100644
--- a/usr/src/cmd/fm/fmdump/common/fault.c
+++ b/usr/src/cmd/fm/fmdump/common/fault.c
@@ -249,13 +249,17 @@ postprocess_msg(char *msg)
 static int
 flt_msg(fmd_log_t *lp, const fmd_log_record_t *rp, FILE *fp)
 {
-	char *msg;
+	char *msg, *uuid = "-", *code = "-";
 
 	if ((msg = fmd_msg_gettext_nv(g_msg, NULL, rp->rec_nvl)) == NULL) {
-		(void) fprintf(stderr, "%s: failed to format message: %s\n",
-		    g_pname, strerror(errno));
+		(void) nvlist_lookup_string(rp->rec_nvl, FM_SUSPECT_UUID,
+		    &uuid);
+		(void) nvlist_lookup_string(rp->rec_nvl, FM_SUSPECT_DIAG_CODE,
+		    &code);
+		(void) fprintf(stderr, "%s: failed to format message for "
+		    "diagcode %s, event %s: %s\n\n", g_pname, code, uuid,
+		    strerror(errno));
 		g_errs++;
-		return (-1);
 	} else {
 		postprocess_msg(msg);
 		fmdump_printf(fp, "%s\n", msg);
